### PR TITLE
Adding bits8 to inductor codegen

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116295
* #116294

Summary:
WIP, just for feedback

Test Plan:
TODO

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by jgong5 — COMMENTED** [review]
Add test?

**Comment by jerryzh168** [comment]
> Add test?

sure, will add some tests after I got feedback from Elias on this

File: c10/core/ScalarType.cpp [hunk 1]
```diff
@@ -61,6 +61,11 @@ ScalarType promoteTypes(ScalarType a, ScalarType b) {
         toString(b));
   }
 
+  if ((a == ScalarType::Bits8 && b == ScalarType::Byte) ||
+      (a == ScalarType::Byte && b == ScalarType::Bits8)) {
+    return ScalarType::Byte;
```
Comment (ezyang) [inline c10/core/ScalarType.cpp:66]: No, no type promotion is allowed for bits

**Comment by ezyang** [comment]
I recommend looking at this stack of PRs https://github.com/pytorch/pytorch/pull/116810 and also PRs I recently landed for uint

There are some fundamental design questions here, first and foremost what underlying c++ type do we represent bits with. Unsigned int is expedient but I am a bit suspicious as it forces interpretation of all bits operations to coincide with unsigned int. But for example, while unsigned ints participate in usual arithmetic conversions, bits definitely SHOULD NOT. So maybe the c10::bits wrappers are better, even if you end up having to build out operators you want.

**Comment by jerryzh168** [comment]
> I recommend looking at this stack of PRs #116810 and also PRs I recently landed for uint
> 
> There are some fundamental design questions here, first and foremost what underlying c++ type do we represent bits with. Unsigned int is expedient but I am a bit suspicious as it forces interpretation of all bits operations to coincide with unsigned int. But for example, while unsigned ints participate in usual arithmetic conversions, bits definitely SHOULD NOT. So maybe the c10::bits wrappers are better, even if you end up having to build out operators you want.

I see, this makes sense, we only want to support a subset of operations (bits operations mostly) of uint8.

**Comment by jerryzh168** [comment]
closing this one since we don't need it anymore after we added https://github.com/pytorch/pytorch/pull/117208

**Pull request closed without merging**